src/compose/ComposeText.js

```diff
@@ -68,8 +68,14 @@ class ComposeText extends React.Component {
     } else if (isTopicNarrow(narrow)) {
       sendMessage(auth, 'stream', narrow[0].operand, narrow[1].operand, text);
     }
+    this.clearInput();
+  }
 
+  clearInput = () => {
     this.textInput.clear();
```

Comment:
The state "text" and input field value at any given point isn't connected in anyway, this line is needed.